core/lib/zksync_core/src/api_server/tx_sender/mod.rs

```diff
@@ -690,6 +693,140 @@ impl TxSender {
+                        .await
+                        .map_err(|err| SubmitTxError::Internal(err.into()))?;
+                    connection
+                        .factory_deps_dal()
+                        .insert_factory_deps(
+                            block_args.resolved_block_number(),
+                            &HashMap::from_iter(vec![(code_hash, code.0.clone())]),
+                        )
+                        .await
+                        .map_err(|err| SubmitTxError::Internal(err.into()))?;
+                }
+
+                if let Some(state) = &overrides.state {
+                    let mut connection = self.acquire_replica_connection().await?;
+
+                    // First find storage keys to clear
+                    let keys_to_clear = connection
+                        .storage_logs_dal()
+                        .dump_all_storage_logs_for_tests()
+                        .await
+                        .iter()
+                        .filter(|log| &log.address == account)
+                        .map(|log| log.key)
+                        .unique()
+                        .collect::<Vec<_>>();
+
+                    // Now clear keys and add new values
+                    let mut logs: Vec<StorageLog> = keys_to_clear
+                        .iter()
+                        .map(|key| {
+                            StorageLog::new_write_log(
+                                StorageKey::new(AccountTreeId::new(*account), *key),
+                                H256::zero(),
+                            )
+                        })
+                        .collect();
+                    logs.extend(state.iter().map(|(key, value)| {
+                        StorageLog::new_write_log(
+                            StorageKey::new(AccountTreeId::new(*account), *key),
+                            *value,
+                        )
+                    }));
+
+                    connection
+                        .storage_logs_dal()
+                        .append_storage_logs(
+                            block_args.resolved_block_number(),
+                            &[(H256::zero(), logs)],
+                        )
+                        .await
+                        .map_err(|err| SubmitTxError::Internal(err.into()))?;
+                }
+
+                if let Some(state_diff) = &overrides.state_diff {
+                    let logs = state_diff
+                        .iter()
+                        .map(|(key, value)| {
+                            StorageLog::new_write_log(
+                                StorageKey::new(AccountTreeId::new(*account), *key),
+                                *value,
+                            )
+                        })
+                        .collect();
+                    self.acquire_replica_connection()
+                        .await?
+                        .storage_logs_dal()
+                        .append_storage_logs(
```

Comment:
So you are trying literally overridy the state in the db. 

1. It's impossible using db replica, because inserting is not possible in replicas
2. You don't even use transactions you will completely mess up the state.

Instead you have to create `StorageView` fill it with necessray data and propagate to vm

Reply:
Changed in e2ad0f4c1690bbe45e1b43268a170c7878b99122